src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -871,16 +871,14 @@ public function addCartRuleAction(int $orderId, Request $request): RedirectRespo
             if ($addOrderCartRuleForm->isValid()) {
                 $data = $addOrderCartRuleForm->getData();
 
-                $invoiceId = $data['apply_on_all_invoices'] ? null : (int) $data['invoice_id'];
-
                 try {
                     $this->getCommandBus()->handle(
                         new AddCartRuleToOrderCommand(
                             $orderId,
                             $data['name'],
                             $data['type'],
                             $data['value'] ?? null,
-                            $invoiceId
+                            (int) $data['invoice_id']
```

Comment:
Be ceraful `$data['invoice_id']` can still be nul when the order has no invoices yet Did ou check if it works fine?
Or should it look like the value field? `(int) $data['invoice_id'] ?? null` .

Reply:
added a safety check just to make sure.